src/main/java/ichttt/mods/mcpaint/common/block/TileEntityCanvas.java

```diff
@@ -97,13 +97,15 @@ public void handleUpdateTag(@Nonnull CompoundTag tag) {
     @Nonnull
     @Override
     public CompoundTag getUpdateTag() {
-        return this.save(new CompoundTag());
+        CompoundTag tag = new CompoundTag();
+        this.saveAdditional(tag);
```

Comment:
did you test in multiplayer that saveAdditional is enough? Cause the old code caused a full save, while this causes a partial update